### ISSUE TITLE
[docdb] Provide the ability to enforce a min version of TLS 

### ISSUE THREAD

**Issue by iSignal** [yugabyte/yugabyte-db#6671]
1. At the docdb level, it looks like it should be a gflag that maps to https://www.openssl.org/docs/man1.1.0/man3/SSL_CTX_set_min_proto_version.html

There's also a few deprecated flags like `SSL_OP_NO_SSLv3, SSL_OP_NO_TLSv1, SSL_OP_NO_TLSv1_1, SSL_OP_NO_TLSv1_2, SSL_OP_NO_DTLSv1, SSL_OP_NO_DTLSv1_2` at https://www.openssl.org/docs/man1.1.0/man3/SSL_CTX_set_options.html for previous versions of openssl.

To verify, one would use something like `openssl s_client -connect <host:port> -no_tls1_2`

2. For postgres, there's an option ssl_min_protocol_version in https://www.postgresql.org/docs/12/runtime-config-connection.html

3.  https://www.java.com/en/configure_crypto.html discusses some system properties used to configure this for Java clients (should be applicable to both cassandra drivers and YB Platform, hopefully) 

Will dig into a bit more. cc @bmatican

**Comment by tylarb** [comment]
@iSignal is this request covering platform (that is, the webserver) as well as connections to database?

On older versions of platform, you can set the TLS minimum version in the nginx conf file.

**Comment by iSignal** [comment]
Yes this is for db RPC (java, C++) and db client (postgres, cql) connections. 

nginx conf in platform is also a good point, I didn't cover that point in the list above.

**Comment by iSignal** [comment]
#6865 added the flag `--ssl_protocols` for RPC (and CQL servers).
